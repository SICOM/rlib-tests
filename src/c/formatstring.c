/*
 *  Copyright (C) 2016 SICOM Systems, INC.
 *
 *  Author: Zoltán Böszörményi <zboszormenyi@sicom.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of version 2 of the GNU General Public
 * License as published by the Free Software Foundation.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public
 * License along with this program; if not, write to the
 * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

#include <config.h>

#include <stdio.h>
#include <rlib.h>

char *data[2][2] = {
	[0][0] = "date",
	[0][1] = "number",

	[1][0] = "2016-01-01 10:27:37",
	[1][1] = "100.0"
};

int main(int argc, char **argv) {
	rlib *r;

	if (argc == 1) {
		printf("usage: %s [ pdf | xml | txt | csv | html ]\n", argv[0]);
		return 1;
	}

	r = rlib_init();
	rlib_add_datasource_array(r, "local_array");
	rlib_add_query_array_as(r, "local_array", data, 2, 2, "data");
	rlib_add_report(r, "formatstring.xml");
	rlib_set_output_format_from_text(r, argv[1]);
	rlib_set_output_parameter(r, "debugging", "yes");
	rlib_set_output_parameter(r, "only_quote_strings", "yes");
	rlib_set_locale(r, "en_US");
	rlib_execute(r);
	rlib_spool(r);
	rlib_free(r);

	return 0;
}
