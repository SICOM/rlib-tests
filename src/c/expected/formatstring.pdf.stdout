%PDF-1.3
%·¾­ª
1 0 obj
<<
/Type /Catalog
/Pages 2 0 R
>>
endobj
2 0 obj
<<
/Type /Pages
/Kids [ 4 0 R ]
/Count 1
>>
endobj
3 0 obj
<<
/Producer (Haru Free PDF Library 2.2.1)
/Title (RLIB Report)
>>
endobj
4 0 obj
<<
/Type /Page
/MediaBox [ 0 0 612 792 ]
/Contents 5 0 R
/Resources <<
/ProcSet [ /PDF /Text /ImageB /ImageC /ImageI ]
/Font <<
/F1 7 0 R
>>
>>
/Parent 2 0 R
/Rotate 0
>>
endobj
5 0 obj
<<
/Length 6 0 R
>>
stream
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
BT
1 0 0 1 14.4 770.76001 Tm
(Legacy format date string: 01 Fri                                                                   ) Tj
ET
BT
1 0 0 1 14.4 761.76001 Tm
(New style date format string: 01 Fri                                                                ) Tj
ET
BT
1 0 0 1 14.4 752.76001 Tm
(Legacy \050valid\051 number format string - no colon - You have 100 apples                                ) Tj
ET
BT
1 0 0 1 14.4 743.76001 Tm
(Legacy \050valid\051 number format string: You have 100 apples                                            ) Tj
ET
BT
1 0 0 1 14.4 734.76001 Tm
(Legacy \050invalid\051 number format string: 100 \045a                                                       ) Tj
ET
BT
1 0 0 1 14.4 725.76001 Tm
(New style \050invalid\051 number format string: 100 \045a                                                    ) Tj
ET
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf
/F1 9 Tf

endstream
endobj
6 0 obj
1056
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Courier
/Subtype /Type1
/Encoding /WinAnsiEncoding
>>
endobj
xref
0 8
0000000000 65535 f
0000000015 00000 n
0000000064 00000 n
0000000123 00000 n
0000000205 00000 n
0000000391 00000 n
0000001501 00000 n
0000001521 00000 n
trailer
<<
/Root 1 0 R
/Info 3 0 R
/Size 8
>>
startxref
1616
%%EOF
